src/main/java/com/google/sps/servlets/InfoCardServlet.java

```diff
@@ -29,16 +35,23 @@
 @WebServlet("/info-cards")
 public class InfoCardServlet extends HttpServlet {
 
-  private static final String BASE_URL = "https://civicinfo.googleapis.com/civicinfo/v2/voterinfo?";
+  private static final String BASE_URL = "https://civicinfo.googleapis.com/civicinfo/v2/voterinfo";
 
+  /**
+   * Makes an API call to voterInfoQuery in the Google Civic Information API and puts Position and
+   * Candidate Entities in Datastore from the response. Finds the chosen Election Entity in the
+   * Datastore and fills is properties with the corresponding API response data.
+   *
+   * TODO: Get Proposition data and add it as a field to the Election Entity.
+   *
+   * @param request the HTTP request containing user address and electionId as parameters
+   * @param response the HTTP response, contains error message if an error occurs
+   */
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-
-    // find the corresponding election and check fields
-    // TODO (anooshree): iterate through Datastore and find the election
-    //                   with the matching electionId
-    // if this election is already populated, we don't need to make another query
-
+  public void doPut(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Read in the user-chosen address and election ID, to be used as parameters to the
+    // voterInfoQuery. If these parameters aren't found, API call can't be performed so return 
```

Comment:
I think you can move this in the javadoc, where you talk about @param request; and then the rest of comment is more of a "what is going on" kind of comment, which we want to avoid